src/components/widgets/EventWidget.tsx

```diff
@@ -1,49 +1,59 @@
 import { useNavigation } from '@react-navigation/core';
 import moment from 'moment';
 import React, { useCallback, useContext, useState } from 'react';
-import { useQuery } from 'react-apollo';
+import { useQuery } from 'react-query';
 
 import { consts, Icon, texts } from '../../config';
-import { graphqlFetchPolicy } from '../../helpers';
-import { useHomeRefresh, useRefreshTime, useVolunteerData } from '../../hooks';
-import { NetworkContext } from '../../NetworkProvider';
+import { useHomeRefresh, useVolunteerData } from '../../hooks';
 import { getQuery, QUERY_TYPES } from '../../queries';
+import { ReactQueryClient } from '../../ReactQueryClient';
 import { SettingsContext } from '../../SettingsProvider';
 import { ScreenName, WidgetProps } from '../../types';
 
 import { DefaultWidget } from './DefaultWidget';
 
-const { REFRESH_INTERVALS, ROOT_ROUTE_NAMES } = consts;
+const { ROOT_ROUTE_NAMES } = consts;
 
 const today = moment().format('YYYY-MM-DD');
 
 export const EventWidget = ({ text, additionalProps }: WidgetProps) => {
   const navigation = useNavigation();
-  const refreshTime = useRefreshTime('event-widget', REFRESH_INTERVALS.ONCE_A_DAY);
-  const { isConnected, isMainserverUp } = useContext(NetworkContext);
   const { globalSettings } = useContext(SettingsContext);
   const { deprecated = {}, hdvt = {} } = globalSettings;
   const { events: showVolunteerEvents = false } = hdvt as { events?: boolean };
-  const [queryVariables] = useState<{ dateRange?: string[]; limit?: number; order: string }>({
-    dateRange: [today, today],
-    order: 'listDate_ASC'
+  const [queryVariables] = useState<{ dateRange?: string[] }>({
+    dateRange: [today, today]
   });
-  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp, refreshTime });
 
   if (additionalProps?.noFilterByDailyEvents) {
     delete queryVariables.dateRange;
   }
 
-  const { data, loading, refetch } = useQuery(
-    getQuery(
+  const {
+    data,
+    isLoading: loading,
+    refetch
+  } = useQuery(
+    [
       deprecated?.events?.listingWithoutDateFragment
         ? QUERY_TYPES.EVENT_RECORDS_WITHOUT_DATE_FRAGMENT
-        : QUERY_TYPES.EVENT_RECORDS
-    ),
+        : QUERY_TYPES.EVENT_RECORDS_COUNT,
+      queryVariables
+    ],
+    async () => {
+      const client = await ReactQueryClient();
+
+      return await client.request(
+        getQuery(
+          deprecated?.events?.listingWithoutDateFragment
+            ? QUERY_TYPES.EVENT_RECORDS_WITHOUT_DATE_FRAGMENT
+            : QUERY_TYPES.EVENT_RECORDS_COUNT
+        ),
+        queryVariables
+      );
+    },
     {
-      fetchPolicy,
-      variables: queryVariables,
-      skip: !refreshTime
+      keepPreviousData: true
```

Comment:
What exactly does `keepPreviousData` do? How long does it keep the data? When new data comes in, does it overwrite it or add it to the old data?

Reply:
ha good part. this is something useful for pagination but not needed here. it can be eliminated i guess.
for pagination it can be used to keep data from a fetch available while new data is being requested.